rest-core/src/Rest/Driver/Routing.hs

```diff
@@ -319,14 +319,15 @@ mkMultiHandler sBy run (GenHandler dict act sec) = GenHandler <$> mNewDict <*> p
               mapExceptT (run i) (act (Env hs ps v))
          return . StringHashMap.fromList $ zipWith (\(k, _) b -> (k, eitherToStatus b)) (StringHashMap.toList vs) bs
 
-mkMultiGetHandler :: forall m s. (Applicative m, Monad m) => Rest.Router m s -> Handler m
-mkMultiGetHandler root = mkInputHandler (xmlJsonI . multipartO) $ \(Resources rs) -> multiGetHandler rs
+mkMultiGetHandler :: forall m s. (Applicative m, Monad m) => Config m -> Rest.Router m s -> Handler m
+mkMultiGetHandler cfg root = mkInputHandler (xmlJsonI . multipartO) $ \(Resources rs) -> multiGetHandler rs
   where
-    multiGetHandler rs = lift $
-      do mapM (runResource root) rs
+    multiGetHandler rs = lift . forM rs $ \r -> do
+      o <- (multiGetHook cfg) r (runResource cfg root r)
```

Comment:
I'm wondering if this is the right way to call the hook. It could also be passed the `root` and `r`, and then have to do the `runResource` itself. That's how I originally thought it would be, though I'm not sure it's better. Then it also wouldn't be called `hook`, but just `multiGet`. It could even include the `forM`, though I'm not sure that helps. What do you think?